Doc/library/typing.rst

```diff
@@ -1035,6 +1040,17 @@ These are not used in annotations. They are building blocks for creating generic
           '''Add two numbers together.'''
           return x + y
 
+   Without ``ParamSpec``, the simplest way to annotate this previously was to
+   use a ``TypeVar`` with bound ``Callable[..., Any]``.  However this causes two
+   problems:
+
+      1. The type checker will type check the inner function, but allow
+         incorrectly typed arguments to be passed into ``add_two``, causing a
+         ``TypeError`` during runtime because the ``...`` tells the type checker
+         to not perform validation on argument types.
```

Comment:
Not so, I think. This is the old solution for decorators that don't change the signature:
```
from typing import *

F = TypeVar("F", bound=Callable[..., Any])

def with_log(f: F) -> F:
    def inner(*args, **kwds):
        print(args, kwds)
        return f(*args, **kwds)
    return inner  # type: ignore

@with_log
def add(a: int, b: int) -> int:
    return a + b

add(1, 2)  # OK
add(2, "")  # Error
```
Two problems with this are:
- no type checking inside inner() -- the best types you can use are all `Any`
- need `# type: ignore` (or `cast()`) on `return inner`

But once the decorator is written and debugged, for the user it's type-safe, because the type variable `F` transmits the signature of the decorated function to the result.

Reply:
Ahh ok thanks for the really helpful explanation!
I misinterpreted it as this at first:
```python
def with_log(f: Callable[..., Any]) -> Callable[..., Any]:
    ...
```
which would allow the invalid call to pass through.